Lib/test/libregrtest/utils.py

```diff
@@ -712,3 +713,19 @@ def get_signal_name(exitcode):
         pass
 
     return None
+
+
+ILLEGAL_XML_CHARS_RE = re.compile(
+    '['
+    '\x00-\x1F'      # ASCII control characters
+    '\uD800-\uDFFF'  # surrogate characters
+    '\uFFFE'
+    '\uFFFF'
+    ']')
+
+def _escape_xml_replace(regs):
+    code_point = ord(regs[0])
+    return f"&#{code_point};"
```

Comment:
No, this does not work. Character references with illegal code are still illegal.

We can either remove invalid characters completely, or use backslash replacing:
```py
def _backslash_replace(text):
    return ''.join(fr'\x{ord(i):02x}' if c <= '\xff' else ascii(c)[1:-1]
                   for c in text)
```

Reply:
Well, it actually "works", but not in the way you may expect. Since `&` is escaped in the XML again, you get back literal `&#...;` after deserialization. It has no advantage over using any other escaping, for example backslash escaping which is more familiar to Python users. In any case this conversion is irreversible.